receiver/receivercreator/factory.go

```diff
@@ -52,17 +52,22 @@ func (f *Factory) CustomUnmarshaler() component.CustomUnmarshaler {
 		}
 		c := intoCfg.(*Config)
 
-		// TODO: Change the sub-receivers to be under "receivers" to allow other config for the main receiver-creator receiver.
-		for subreceiverKey := range sourceViperSection.AllSettings() {
-			cfgSection := sourceViperSection.GetStringMap(subreceiverKey + "::config")
+		if err := sourceViperSection.Unmarshal(&c); err != nil {
+			return err
+		}
+
+		receiversCfg := viperSub(sourceViperSection, "receivers")
+
+		for subreceiverKey := range receiversCfg.AllSettings() {
+			cfgSection := receiversCfg.GetStringMap(subreceiverKey + "::config")
```

Comment:
What does "::" do here? Can you add a comment that describes why we are concatenating the key this way?

Reply:
I'll use viperSub as well so it's more clear.